lxc/init.go

```diff
@@ -370,6 +381,39 @@ func (c *cmdInit) create(conf *config.Config, args []string) (lxd.InstanceServer
 	// Validate the network setup
 	c.checkNetwork(d, name)
 
+	if deviceOverrides != nil {
```

Comment:
hey @markylaing  I've not had chance yet to look at this in detail, but this doesn't look like the correct approach.
If I'm reading this correctly, you're updating the instance's config after it has been created.
I think we need to override the config and then include it in the instance creation request itself.

Reply:
I'm confused. This is what I was asking here https://github.com/lxc/lxd/issues/10796#issuecomment-1288924934

I can't get the expanded devices until the instance has been created... I'd need to check which profiles are going to be applied in advance.

Reply:
Yeah I can see why you got confused.

What we meant is that is effectively what needs to be done, but I need to look at the API to see if its possible for us to "effectively" do at create time.

Reply:
Ok. We need to query for the profiles that are to be applied in advance, then override any matching devices passed via the `--device` flag. (That was option 1 in https://github.com/lxc/lxd/issues/10796#issuecomment-1288924934)

Reply:
yeah potentially we need to "expand" the provisional device config and then copy and apply changes to the specific device(s) to send as the device config for the new instance.

Reply:
Ok I think I've made the changes.